pyresample/future/spherical/extent.py

```diff
@@ -0,0 +1,300 @@
+        raise ValueError("An extent can not be defined by a line.")
+
+
+def _check_valid_extent(extent, use_radians=False):
+    """Check lat/lon extent validity."""
+    # Check extent length
+    if len(extent) != 4:
+        raise ValueError("'extent' must have length 4: [lon_min, lon_max, lat_min, lat_max].")
+    # Convert to np.array
+    extent = np.array(extent)
+    # Check order and values
+    _check_extent_order(extent)
+    _check_extent_values(extent, use_radians=use_radians)
+    # Check is not point or line
+    _check_is_not_point_extent(extent)
+    _check_is_not_line_extent(extent)
+    # Ensure is a list
+    extent = extent.tolist()
+    return extent
+
+
+def _check_non_overlapping_extents(polygons):
+    """Check that the extents polygons are non-overlapping.
+
+    Note: Touching extents are considered valids.
+    """
+    for poly in polygons:
+        # Intersects includes within/contain/equals/touches !
+        n_intersects = np.sum([p.intersects(poly) and not p.touches(poly) for p in polygons])
+        if n_intersects >= 2:
+            raise ValueError("The extents composing SExtent can not be duplicates or overlapping.")
+
+
+def _is_global_extent(polygons):
+    """Check if a list of extents composes a global extent."""
+    from shapely.geometry import Polygon
+    from shapely.ops import unary_union
+
+    # Try union the polygons
+    unioned_polygon = unary_union(polygons)
+    # If still a MultiPolygon, not a global extent
+    if not isinstance(unioned_polygon, Polygon):
+        return False
+    # Define global polygon
+    global_polygon = Polygon.from_bounds(-180, -90, 180, 90)
+    # Assess if global
+    is_global = global_polygon.equals(unioned_polygon)
+    return is_global
+
+
+class SExtent(object):
+    """Spherical Extent.
+
+    SExtent longitudes are defined between -180 and 180 degree.
+    Geometrical operations are performed on a planar coordinate system.
+
+    A spherical geometry crossing the anti-meridian will have an SExtent
+     composed of [lon_start, 180, ...,...] and [-180, lon_end, ..., ...]
+
+    Important notes:
+    - SExtents touching at the anti-meridian are considered to not touching !
+    - SExtents.intersects predicate includes also contains/within cases.
+    - In comparison to shapely polygon behaviour, SExtent.intersects is
+      False if SExtents are just touching.
+
+    There is not an upper limit on the number of extents composing SExtent !!!
+    The only conditions is that the extents do not intersect/overlap each other.
+
+    More specifically, extents composing an SExtent:
+    - must represent a polygon (not a point or a line),
+    - can touch each other,
+    - but can not intersect/contain/overlap each other.
+
+    Examples of valid SExtent definitions:
+      extent = [x_min, x_max, y_min, ymax]
+      sext = SExtent(extent)
+      sext = SExtent([extent, extent])
+      sext = SExtent(extent, extent, ...)
```

Comment:
Is there a need or benefit to supporting both of these types of multi-extent inputs?

Reply:
No need. Which one would you suggest to drop? 
I would keep the first.

Reply:
Sorry, it's been a long time since I've looked at this and I can't quite remember what this class does. So it merges extents (bounding box?) into a single MultiPolygon? The idea being you either have one contiguous extent or you have two extents that "cross" the antimeridian, right? Will there ever be a need for more than two extents?

Reply:
Yes. One could discretize a SPolygon in lot of non-intersecting extents ... 
But the main use case is to deal with the antimeridian ...

Reply:
I'm not sure what I prefer. Initially I was going to say only use the second one (passing a list), but since the usual case is a single 4-element extent then maybe the first and the third should be the only ways to do it.